### PR TITLE
Make createContext return 0 on failure

### PR DESCRIPTION
**Pull request by matanui159**
I noticed that in `library_egl.js` on line 369 that it will fail if `createContext` returns 0. However, checking the `createContext` code shows that it would instead return `null` (the return value of `getContext`) if it fails to create a context making EGL think that it was successful and then producing weird errors later.

As a side note, since unlike the previous PR, this one's copyright isn't owned by Clipchamp, is there a recommended way to represent that in the `AUTHORS` list? I don't mind too much just thought I might ask.

### CODE REVIEW

**Comment by sbc100** [comment]
Looks like you are correct.   I wonder as a rule if we should be using less precise checks such as `if (EGL.context)` to avoid this kind of distinction, but this change looks good based on the current code.

I also wonder why there appears to be no test coverage for this case?  Don't we have tests that fail to create contexts?

**Comment by matanui159** [comment]
> As a side note, since unlike the previous PR, this one's copyright isn't owned by Clipchamp, is there a recommended way to represent that in the AUTHORS list? I don't mind too much just thought I might ask.

Is there an answer to this question?

> I also wonder why there appears to be no test coverage for this case? Don't we have tests that fail to create contexts?

Do you want me to write some tests as part of this PR? I am unsure how to write browser tests for emscripten but if you can point me to some documentation I have happy to look into it when I get a chance (might take a while though since I'm a bit busy)

**Comment by kripken** [comment]
@matanui159 

> As a side note, since unlike the previous PR, this one's copyright isn't owned by Clipchamp, is there a recommended way to represent that in the AUTHORS list? I don't mind too much just thought I might ask.

If it matters to you then using a different email, and also a different entry in the AUTHORS file, might be one option. I don't think it matters too much to us for a one-line change. But for anything big we'd need to think about it more I suspect.

@sbc100 

> Don't we have tests that fail to create contexts?

I don't think we do currently.

I think it would be good to add a test for that. @matanui159 for writing tests, it's best to look at the existing ones. For example, `tests/glgetattachedshaders.c` is used when running `tests/runner.py browser.test_glgetattachedshaders` (see `tests/test_browser.py`). Maybe the test could use an EM_ASM to make the creation fail somehow, but I'm not sure offhand. If you get stuck or need help let us know!

**Comment by matanui159** [comment]
@kripken 

> If it matters to you then using a different email, and also a different entry in the AUTHORS file, might be one option. I don't think it matters too much to us for a one-line change. But for anything big we'd need to think about it more I suspect.

It doesn't really matter to me that much so I probably won't bother with it

> I think it would be good to add a test for that. @matanui159 for writing tests, it's best to look at the existing ones. For example, tests/glgetattachedshaders.c is used when running tests/runner.py browser.test_glgetattachedshaders (see tests/test_browser.py). Maybe the test could use an EM_ASM to make the creation fail somehow, but I'm not sure offhand. If you get stuck or need help let us know!

I'll look into adding a test if I get a chance this afternoon. I haven't had much luck with running browser-based tests locally but even if I can't get it to run I could always try trial and error with each commit :stuck_out_tongue:

File: tests/test_browser.py [hunk 1]
```diff
@@ -1576,6 +1576,13 @@ def test_egl_width_height(self):
   def test_egl_width_height_with_proxy_to_pthread(self):
     self._test_egl_width_height_base('-s', 'USE_PTHREADS', '-s', 'PROXY_TO_PTHREAD=1')
 
+  @requires_graphics_hardware
+  def test_egl_createcontext_error(self):
+    create_test_file('test_egl_createcontext_error.c', self.with_report_result(open(path_from_root('tests', 'test_egl_createcontext_error.c')).read()))
```
Comment (kripken) [inline tests/test_browser.py]: I think these 3 lines can be replaced with `self.btest('test_egl_createcontext_error.c', '1', args=['-lEGL', '-lGL'])`

**Comment by matanui159** [comment]
What's the current status for this PR?

**Comment by kripken** [comment]
Perfect, thanks @matanui159!

(sorry for the delay, I was on vacation)

**Pull request merged**